### PR TITLE
Implemented Validation of Collection Name Length Issue 10

### PR DESCRIPTION
**Pull request by JimPriestley**
Character length 1-15 characters
  Added Tests for impacted resources
  Updated Documentaion
  Added Missing new-line at end of .gitignore and the common PSM

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xremotedesktopsessionhost/13)

### CODE REVIEW

**Comment by JimPriestley** [comment]
This is my first ever pull request, so I picked something simple. Please let me know if I'm missing anything.

**Comment by kwirkykat** [comment]
@JimPriestley Closing and re-opening the PR to try to get the CLA bot to re-trigger.

**Comment by JimPriestley** [comment]
Reopening to clear CLA-Required now that I joined the MS GitHub Org

File: .gitignore [hunk 1]
```diff
@@ -1 +1 @@
-DscResource.Tests/
+DscResource.Tests/
```
Comment (TravisEz13) [inline .gitignore]: Why `^M`  The file should end like any other file

File: README.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ This path resolves to the same location as the value of the FilePath parameter,
 * **FolderName**: Specifies the name of the folder that the RemoteApp program appears in on the Remote Desktop Web Access (RD Web Access) webpage and in the Start menu for subscribed RemoteApp and Desktop Connections.
 * **CommandLineSetting**: Specifies whether the RemoteApp program accepts command-line arguments from the client at connection time.
 * **RequiredCommandLine**: Specifies a string that contains command-line arguments that the client can use at connection time with the RemoteApp program.
-* **IconIndex**: Specifies the index within the icon file (specified by the IconPath parameter) where the RemoteApp program�s icon can be found.
+* **IconIndex**: Specifies the index within the icon file (specified by the IconPath parameter) where the RemoteApp program�s icon can be found.
```
Comment (TravisEz13) [inline README.md]: Please fix the apostrophe in `program's`

File: .gitignore (hunk 1 above)
Comment (JimPriestley) [inline .gitignore]: I'm stuck here, can you help?
All I did was add a newline to meet the style guide requirements.
I have tried deleting and recreating this now about 10 times, and the ^M returns every time.
Is there a way to just exclude this file from the PR, and revert to the original?

**Comment by JimPriestley** [comment]
I'm going to close this PR and create a new fork and PR to solve for the broken .gitignore file.

**Pull request closed without merging**